test/extended/baremetal/hosts.go

```diff
@@ -186,4 +186,78 @@ var _ = g.Describe("[sig-installer][Feature:baremetal][Serial] Baremetal platfor
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(found).To(o.BeFalse())
 	})
+
+	g.It("configure BIOS settings during cleaning", func() {
+		var procTurboMode = "ProcTurboMode"
+
+		skipIfNotBaremetal(oc)
+
+		dc := oc.AdminDynamicClient()
+		hostData, secretData := helper.GetExtraWorkerData(0)
+
+		// Deploy extra worker and wait
+		host, _ := helper.CreateExtraWorker(hostData, secretData)
+		host = helper.WaitForProvisioningState(host, "available")
+		hostName := getStringField(*host, "baremetalhost", "metadata.name")
+
+		hfsClient := hostfirmwaresettingsClient(dc)
+		hfs := getHostFirmwareSettings(hfsClient, hostName)
+
+		status, _, err := unstructured.NestedStringMap(hfs.Object, "status", "settings")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		spec, _, err := unstructured.NestedStringMap(hfs.Object, "spec", "settings")
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// Change HostFirmwareSetting to an invalid value
+		spec[procTurboMode] = "Foo"
```

Comment:
Just question to learn. Can `spec` be null?, or if it is null `unstructured.NestedStringMap` returns error?

Reply:
It should not be null as its allocated on the create of the resource - https://github.com/metal3-io/baremetal-operator/blob/main/controllers/metal3.io/baremetalhost_controller.go#L1372.

I can add a check for nil.